components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,240 @@
+                .eventName(event.getEventName())
+                .request((HttpServletRequest)properties.get("request"))
+                .eventParams(params)
+                .authenticationContext((AuthenticationContext)properties.get("context"))
+                .authenticatorStatus((AuthenticatorStatus)properties.get("authenticationStatus"))
+                .authenticatedUser(authenticatedUser)
+                .loginIdentifier(loginIdentifier)
+                .build();
+    }
+
+    /**
+     * Retrieve the audience.
+     *
+     * @param eventUri     Event URI.
+     * @return Audience string.
+     */
+    public static SecurityEventTokenPayload buildSecurityEventToken(EventPayload eventPayload,
+                                                                    AuthenticationContext context, String eventUri)
+            throws IdentityEventException {
+
+        if (eventPayload == null) {
+            throw new IdentityEventException("Invalid event payload input: Event payload input cannot be null.");
+        }
+
+        if (StringUtils.isEmpty(eventUri)) {
+            throw new IdentityEventException("Invalid event URI input: Event URI input cannot be null or empty.");
+        }
+
+        SecurityEventTokenPayload securityEventTokenPayload = new SecurityEventTokenPayload();
+        securityEventTokenPayload.setIss(getReference("", ""));
+        securityEventTokenPayload.setIat(System.currentTimeMillis());
+        securityEventTokenPayload.setJti(UUID.randomUUID().toString());
+        securityEventTokenPayload.setRci(getCorrelationID());
+        Map<String, EventPayload> eventMap = new HashMap<>();
+        eventMap.put(eventUri, eventPayload);
+        securityEventTokenPayload.setEvent(eventMap);
+        return securityEventTokenPayload;
+    }
+
+    /**
+     * Get correlation id from the MDC.
+     * If not then generate a random UUID, add it to MDC and return the UUID.
+     *
+     * @return Correlation id
+     */
+    public static String getCorrelationID() {
+
+        String correlationID = MDC.get(CORRELATION_ID_MDC);
+        if (StringUtils.isBlank(correlationID)) {
+            correlationID = UUID.randomUUID().toString();
+            MDC.put(CORRELATION_ID_MDC, correlationID);
+        }
+        return correlationID;
+    }
+
+    private static void setLocalUserClaimsToAuthenticatedUser(AuthenticatedUser authenticatedUser,
+                                                       AuthenticationContext context) {
+
+        Map<String, String> claimMappings = (Map<String, String>) context.getParameters()
+                    .get(Constants.SP_TO_CARBON_CLAIM_MAPPING);
+
+            if (claimMappings == null) {
+                log.debug("No local claim mappings found for the authenticated user from the context.");
+                return;
+            }
+
+            Map<ClaimMapping, String> userAttributes = authenticatedUser.getUserAttributes();
+
+            if (userAttributes == null) {
```

Comment:
Hope you will refactor this

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/6e511b91f52c4b71ccff73c989894c76b285ebf0